test/fixtures/azure-fixture/src/main/java/fixture/azure/AzureHttpFixture.java

```diff
@@ -95,10 +107,27 @@ public String toString() {
         };
     }
 
-    public AzureHttpFixture(Protocol protocol, String account, String container, Predicate<String> authHeaderPredicate) {
+    public AzureHttpFixture(
+        Protocol protocol,
+        String account,
+        String container,
+        @Nullable String tenantId,
+        @Nullable String clientId,
+        Predicate<String> authHeaderPredicate
+    ) {
+        if (tenantId != null) {
+            if (protocol != Protocol.HTTPS) {
+                fail(null, "when [tenantId] is set, protocol must be HTTPS");
+            }
+            if (clientId == null) {
+                fail(null, "when [tenantId] is set, [clientId] must also be set");
+            }
```

Comment:
Let's assert that `(clientId == null) == (tentantId == null)` since they must both be present or absent together.